src/App.tsx

```diff
@@ -48,31 +61,91 @@ const App = (): JSX.Element => {
+            <Suspense fallback={null}>
+              <button
+                className="absolute right-0 z-20 p-4"
+                onClick={() => setDevtoolsOpen((old) => !old)}
+              >
+                <WrenchScrewdriverIcon className="h-10 fill-grey-primary"></WrenchScrewdriverIcon>
+              </button>
+              {devtoolsOpen && (
+                <Resizable
+                  enable={{ bottom: true }}
+                  defaultSize={{ height: devToolsMinHeight, width: "100%" }}
+                  className="fixed z-10 grid grid-cols-[minmax(0,2fr)_1fr]"
+                  minHeight={devToolsMinHeight}
+                  maxHeight={devToolsMaxHeight}
+                >
+                  <ReactQueryDevtoolsPanel
+                    className="z-20"
+                    style={{ height: "inherit", maxHeight: devToolsMaxHeight }}
+                    setIsOpen={setDevtoolsOpen}
+                    onDragStart={(_) => {}}
+                  />
+                  <DevMenu opts={devOpts} set={setDevOpts} />
+                </Resizable>
+              )}
+            </Suspense>
+          )}
           <Routes>
             <Route path="/" element={<Navigate to="/sessions" />} />
             <Route path="/sessions">
               <Route index element={<ChooseSession />} />
-              <Route path=":sessionId" element={<Edit />} />
+              <Route path=":sessionId" element={<Edit {...devOpts} />} />
             </Route>
             <Route path="*" element={<NoMatch />} />
           </Routes>
-          <ReactQueryDevtools initialIsOpen position="top-left" />
-          {showDevtools && (
-            <Suspense fallback={null}>
-              <ReactQueryDevtoolsProduction position="top-left" />
-            </Suspense>
-          )}
         </QueryClientProvider>
       </BrowserRouter>
     </CookiesProvider>
   );
 };
 
+const DevMenu = (p: { opts: DevOptions; set: (opts: DevOptions) => void }) => (
+  <div className="bg-blue-primary pl-1 text-white-primary">
+    <div>
+      <input
+        type="checkbox"
+        id="showIDs"
+        checked={p.opts.showIDs}
+        onChange={(e) => p.set({ ...p.opts, showIDs: e.target.checked })}
+        className="mr-1"
+      />
+      <label htmlFor="showIDs">show node IDs</label>
+    </div>
+    <div>
+      <input
+        type="checkbox"
+        id="alwaysShowLabels"
+        checked={p.opts.alwaysShowLabels}
+        onChange={(e) =>
+          p.set({ ...p.opts, alwaysShowLabels: e.target.checked })
+        }
+        className="mr-1"
+      />
+      <label htmlFor="alwaysShowLabels">always show labels</label>
+    </div>
+    <div>
+      <input
+        type="checkbox"
+        id="inlineLabels"
+        checked={p.opts.inlineLabels}
+        onChange={(e) => p.set({ ...p.opts, inlineLabels: e.target.checked })}
+        className="mr-1"
+      />
+      <label htmlFor="inlineLabels">inline labels</label>
+    </div>
+  </div>
+);
+
```

Comment:
I would prefer that this were a separate component, not inlined in App, but I can take care of that refactoring later.